pkg/asset/installconfig/powervs/client.go

```diff
@@ -416,8 +416,15 @@ func (c *Client) GetDNSInstancePermittedNetworks(ctx context.Context, dnsID stri
 }
 
 // CreateDNSRecord Creates a DNS CNAME record in the given base domain and CRN.
-func (c *Client) CreateDNSRecord(ctx context.Context, crnstr string, baseDomain string, hostname string, cname string) error {
-	logrus.Debugf("CreateDNSRecord: crnstr = %s, hostname = %s, cname = %s", crnstr, hostname, cname)
+func (c *Client) CreateDNSRecord(ctx context.Context, publish string, crnstr string, baseDomain string, hostname string, cname string) error {
+	if publish == string(types.InternalPublishingStrategy) {
+		return c.createPrivateDNSRecord(ctx, crnstr, baseDomain, hostname, cname)
+	}
+	return c.createPublicDNSRecord(ctx, crnstr, baseDomain, hostname, cname)
```

Comment:
```suggestion
	switch publish {
	case types.InternalPublishingStrategy:
		return c.createPrivateDNSRecord(ctx, crnstr, baseDomain, hostname, cname)
	case types.ExternalPublishingStrategy:
	    return c.createPublicDNSRecord(ctx, crnstr, baseDomain, hostname, cname)
	default:
	    return fmt.Errorf("publish strategy %q not supported", publish)
	}
```

Reply:
I agree with using a case statement but I think not ending with a return will fail golint